### PR TITLE
feature: remove users from issues after 3 months inactivity

### PR DESCRIPTION
**Pull request by za**
https://github.com/OWASP/wrongsecrets/issues/1016

 

### What kind of changes does this PR include?

-   [ ] Fixes or refactors
-   [ ] A new challenge
-   [ ] Additional documentation
-   [x] Something else

#### Description

Automation: remove users from issues after 3 months inactivity

### Checklist:

-   [ ] All the contributions made are solely the work of me and my co-authors
-   [ ] I tested the changes in this PR (if applicable)
-   [ ] I added unit tests to ensure my change works (when change in Java or on front-end code)
-   [ ] I added UI tests to ensure my UI changes work (when change in the overall UI, not needed if just adding a challenge)
-   [ ] The PR passes pre-commit hooks and automated tests

### CODE REVIEW

File: .github/workflows/github-actions-demo.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+      - run: echo "🐧 This job is now running on a ${{ runner.os }} server hosted by GitHub!"
+      - run: echo "🔎 The name of your branch is ${{ github.ref }} and your repository is ${{ github.repository }}."
+      - name: Check out repository code
+        uses: actions/checkout@v4
+      - run: echo "💡 The ${{ github.repository }} repository has been cloned to the runner."
+      - run: echo "🖥️ The workflow is now ready to test your code on the runner."
+      - name: List files in the repository
+        run: |
+          ls ${{ github.workspace }}
+      - run: echo "🍏 This job's status is ${{ job.status }}."
```
Comment (commjoen) [inline .github/workflows/github-actions-demo.yml:18]: not sure if we need this action. Maybe we can remove it?

File: .github/workflows/project-management.yml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+  unassign-issues-labeled-waiting-for-contributor-after-1-days-of-inactivity:
+    name: Unassign issues labeled "Stale" after 1 days of inactivity.
+    if: github.event_name == 'schedule'
+    runs-on: ubuntu-latest
+    steps:
+      - uses: boundfoxstudios/action-unassign-contributor-after-days-of-inactivity@main
+        with:
+          last-activity: 1 # After how many days the issue should get unassigned
+          labels: 'Stale' # Only search for issues containing this labels (comma-separated)
```
Comment (commjoen) [inline .github/workflows/project-management.yml:18]: can we search for all "help wanted" labelled issues instead please

File: .github/workflows/github-actions-demo.yml (hunk 1 above)
Comment (za) [reply .github/workflows/github-actions-demo.yml:18]: Hi @commjoen sure, we can remove this. I am planning to re-create the PR as there are a lot of un-necessary commits on this branch :grin:

File: .github/workflows/project-management.yml (hunk 2 above)
Comment (za) [reply .github/workflows/project-management.yml:18]: > can we search for all "help wanted" labelled issues instead please

Noted.

File: .github/workflows/project-management.yml [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  # Best used in combination with actions/stale to assign a Stale label
+
+  unassign-issues-labeled-waiting-for-contributor-after-1-days-of-inactivity:
+    name: Unassign issues labeled "Stale" after 1 days of inactivity.
+    if: github.event_name == 'schedule'
+    runs-on: ubuntu-latest
+    steps:
+      - uses: boundfoxstudios/action-unassign-contributor-after-days-of-inactivity@main
+        with:
+          last-activity: 1 # After how many days the issue should get unassigned
```
Comment (commjoen) [inline .github/workflows/project-management.yml:17]: maybe we can set this to 90 :) ?

File: .github/workflows/project-management.yml [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+          labels: 'Stale' # Only search for issues containing this labels (comma-separated)
+          exempt-assignees: 'Octo,Cat' # Exempt some assignees from being unassigned
+          labels-to-remove: 'Stale' # Labels to remove after unassigning an issue
```
Comment (commjoen) [inline .github/workflows/project-management.yml:20]: no label to remove i think?

File: .github/workflows/schedule.yml [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+
+on:
+  schedule:
+    - cron: "0 * * * *"
+
+jobs:
+  Run-GitHub-Actions-Schedule:
+    runs-on: ubuntu-latest
+    steps:
+      - run: echo "🎉 Runs every 1 hour"
```
Comment (commjoen) [inline .github/workflows/schedule.yml:12]: this file can possibly remove that as well :)

File: .github/workflows/project-management.yml (hunk 2 above)
Comment (za) [reply .github/workflows/project-management.yml]: Updated

File: .github/workflows/project-management.yml (hunk 3 above)
Comment (za) [reply .github/workflows/project-management.yml]: Fixed

**Comment by za** [comment]
Hi @commjoen can you review the PR? It's ready for review now. BTW, can you squash the commits on this branch into single commit before merging into master?

**Comment by commjoen** [comment]
Hi @za ! I can review, but not squash... sorry

**Comment by commjoen** [comment]
Feel free to squash and force push

**Comment by za** [comment]
In that case, I'll just create a new branch :-)

**Comment by za** [comment]
PR moved to https://github.com/OWASP/wrongsecrets/pull/1050

**Pull request closed without merging**